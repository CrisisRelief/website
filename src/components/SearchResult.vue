<template>
  <div class="search-result-container">
    <hr />
    <a v-if="link.length > 0" :href="link">
      <h3>{{ title }}</h3>
    </a>
    <h3 v-else>{{ title }}</h3>
    <div class="search-result-meta">
      <div class="search-result-categories">
        <span v-for="(category, index) in categories" :key="index" class="search-result-category">
          {{ category }}
          <span v-if="index+1 < categories.length">></span>
        </span>
      </div>
      <div class="search-result-location">{{ location }}</div>
    </div>
    <div class="search-result-description" v-html="description"></div>
    <table
      v-if="Object.entries(contact).length"
      class="search-result-contact table table-borderless my-2"
    >
      <tr v-for="(item, key, index) in contact" :key="index">
        <a :href="item.href">
          <td class="py-0 px-2 h5">
            <FontAwesomeIcon :icon="item.icon" />
          </td>
          <td class="py-0 px-2">{{ item.human }}</td>
        </a>
      </tr>
    </table>
  </div>
</template>

<script>
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";

export default {
  props: {
    title: {},
    category: {},
    tags_flat: {},
    location: {},
    description: {},
    phone: {},
    email: {},
    address: {},
    link: {
      default: ""
    }
  },
  components: { FontAwesomeIcon },
  computed: {
    categories() {
      return [this.category, this.tags_flat].filter(elem => {
        return elem !== undefined && elem.length > 0;
      });
    },
    contact() {
      return ["phone", "email", "address", "link"].reduce((contact, attr) => {
        var human = this[attr];
        var href = human;
        var icon;
        if (human) {
          if (attr == "phone") {
            icon = "phone-square-alt";
            href = "tel:" + human.replace(/[^0-9]/g, "");
          } else if (attr == "link") {
            icon = "address-card";
            human = "Visit Website";
          } else if (attr == "email") {
            icon = "envelope-square";
            href = "mailto:" + human;
          } else if (attr == "address") {
            icon = "directions";
            href = "http://maps.google.com/?q=" + encodeURI(human);
            human = "Get Directions";
          }
          contact[attr] = { href, human, icon };
        }
        return contact;
      }, {});
    }
  }
};
</script>

<style scoped>
h1 {
  font-size: 20px;
  font-weight: 600; /* semibold */
  text-transform: uppercase;
}

.search-result-meta,
.search-result-contact {
  font-weight: 600; /* semibold */
  word-wrap: break-word;
}
</style>
