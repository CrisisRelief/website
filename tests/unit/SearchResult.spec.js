import { shallowMount } from "@vue/test-utils";
import SearchResult from "../../src/components/SearchResult.vue";

const wrapper = shallowMount(SearchResult);

describe("SearchResult", () => {
    it("should match last snapshot", () => {
      expect(wrapper.element).toMatchSnapshot();
    });
});