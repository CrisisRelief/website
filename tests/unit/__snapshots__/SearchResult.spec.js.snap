// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult should match last snapshot 1`] = `
<div
  class="search-result-container"
>
  <hr />
   
  <h3>
    
  </h3>
   
  <div
    class="search-result-meta"
  >
    <div
      class="search-result-categories"
    />
     
    <div
      class="search-result-location"
    >
      
    </div>
  </div>
   
  <div
    class="search-result-description"
  />
   
  <!---->
</div>
`;
